### PR TITLE
Small fixes for Gold/Math section

### PR DESCRIPTION
**Pull request by ChuangSheep**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: content/4_Gold/Combinatorics.mdx [hunk 1]
```diff
@@ -236,15 +266,13 @@ ll fac[MAXN + 1];
...
-ll exp(ll x, ll y, ll p) {
-	ll res = 1; x %= p;
-	while (y) {
-		if (y & 1) {
-			res *= x; res %= p;
-		}
-		x *= x;
-		x %= p;
-		y >>= 1;
+ll binpow(ll x, ll n, ll m) {
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:269]: just curious, why `binpow` instead of `exp`?

Comment (ChuangSheep) [reply content/4_Gold/Combinatorics.mdx:269]: u refered to this implementation in the last pr as 'the' standard so.., i personally also prefer exp

Comment (SansPapyrus683) [reply content/4_Gold/Combinatorics.mdx:269]: you can replace all `binpow`s with `exp`, i just want consistency really
sorry if i failed to communicate properly

**Review by envyaims — APPROVED** [review]
ngl u guys are picky

**Comment by ryanchou-dev** [comment]
BRO I WAS REVIEWING DFDFAFDSF

my timing is impeccable

**Pull request merged**